internal/backend/auth/authloginhttpsvc/svc.go

```diff
@@ -0,0 +1,118 @@
+package authloginhttpsvc
+
+import (
+	_ "embed"
+	"fmt"
+	"net/http"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/auth/authcontext"
+	"go.autokitteh.dev/autokitteh/internal/backend/auth/authloginhttpsvc/web"
+	"go.autokitteh.dev/autokitteh/internal/backend/auth/authsessions"
+	"go.autokitteh.dev/autokitteh/internal/backend/auth/authtokens"
+	"go.autokitteh.dev/autokitteh/internal/backend/db"
+	"go.autokitteh.dev/autokitteh/internal/backend/muxes"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+)
+
+type Deps struct {
+	fx.In
+
+	Muxes    *muxes.Muxes
+	Z        *zap.Logger
+	Cfg      *Config
+	DB       db.DB
+	Sessions authsessions.Store
+	Tokens   authtokens.Tokens
+}
+
+type svc struct {
+	Deps
+}
+
+func Init(deps Deps) error {
+	svc := &svc{
+		Deps: deps,
+	}
+
+	return svc.registerRoutes(deps.Muxes)
+}
+
+func (a *svc) registerRoutes(muxes *muxes.Muxes) error {
+	if a.Cfg.GoogleOAuth.Enabled {
+		if err := registerGoogleOAuthRoutes(muxes.Root, a.Deps.Cfg.GoogleOAuth, a.newSuccessLoginHandler); err != nil {
+			return err
+		}
+	}
+
+	if a.Cfg.GithubOAuth.Enabled {
+		if err := registerGithubOAuthRoutes(muxes.Root, a.Cfg.GithubOAuth, a.newSuccessLoginHandler); err != nil {
+			return err
+		}
+	}
+
+	if a.Cfg.Descope.Enabled {
```

Comment:
I think we should not allow Descope and github/google simulatnously

Reply:
hmmm... i see no technical reason why to forbid it. it's just another mechanism you can choose if it's configured.

Reply:
done